File: cracking_the_coding_interview/1.2permutation.vb
Module Solution
    Sub Main()
        Console.WriteLine (Permutaion("abc", "cba"))
    End Sub
    
    Function Permutaion(Str1 As String, Str2 As String) As Boolean
        Dim char1Array() As Char = Str1.ToCharArray   
        Dim char2Array() As Char = Str2.ToCharArray 
        Array.Sort(char1Array)
        Array.Sort(char2Array)
        
        If (New String(char1Array)).Equals(New String(char2Array))
            Permutaion = True
        Else
            Permutaion = False
        End If
    End Function
End Module

File: cracking_the_coding_interview/1.2permutation.js
function permutation(str1, str2) {
  var sortedStr1 = str1.split('').sort().toString();
  var sortedStr2 = str2.split('').sort().toString();
 
  if (!sortedStr1.localeCompare(sortedStr2)) return true;
  
  return false;
  
  
}

console.log(permutation("abc", "cba"));


